tools/java/org/unicode/cldr/util/VoteResolver.java

```diff
@@ -110,9 +111,53 @@ public static Status fromString(String source) {
      * weight.
      */
     public enum Level {
-        locked(0, 999), street(1, 10), anonymous(0, 8), vetter(4, 5), expert(8, 3), manager(4, 2), tc(20, 1), admin(100, 0);
-        private int votes;
-        private int stlevel;
+        locked(   0 /* votes */, 999 /* stlevel */),
+        street(   1 /* votes */, 10  /* stlevel */),
+        anonymous(0 /* votes */, 8   /* stlevel */),
+        vetter(   4 /* votes */, 5   /* stlevel */),
+        expert(   8 /* votes */, 3   /* stlevel */),
+        manager(  4 /* votes */, 2   /* stlevel */),
+        tc(      20 /* votes */, 1   /* stlevel */),
+        admin(  100 /* votes */, 0   /* stlevel */);
+
+        /**
+         * PERMANENT_VOTES is used by TC voters to "lock" locale+path permanently (including future versions, until unlocked),
+         * in the current VOTE_VALUE table. It is public for STFactory.java and PermanentVote.java.
+         */
+        public static final int PERMANENT_VOTES = 1000;
+
+        /**
+         * LOCKING_VOTES is used (nominally by ADMIN voter, but not really by someone logged in as ADMIN, instead
+         * by combination of two PERMANENT_VOTES) to "lock" locale+path permanently in the LOCKED_XPATHS table.
+         * It is public for STFactory.PerLocaleData.loadVoteValues.
+         */
+        public static final int LOCKING_VOTES = 2000;
+
+        /*
+         * Set voteCountMenu for admin and tc in this static block, which will be run after
+         * all the constructors have run, rather than in the constructor itself, since, for example,
+         * vetter.votes needs to be defined before we can set admin.voteCountMenu.
+         */
+        static {
+            admin.voteCountMenu = new Integer[] {vetter.votes, admin.votes}; /* Not LOCKING_VOTES; see canVoteWithCount */
```

Comment:
You _really_ don't want to do this, since arrays are mutable and thus dangerous, and you expose them with 

        /**
         * Get the array of different vote counts a user of this level can vote with
         *
         * @return the array, or null if the user has no choice of vote count
         */
        public Integer[] getVoteCountMenu() {
            return voteCountMenu;
        }

If the main purpose for containment, then it is also faster to just make them an ImmutableSet. Note that an Immutable set maintains the order (if you care about that).

Then 
            Integer[] menu = getVoteCountMenu();
            if (menu != null) {
                return Arrays.asList(menu).contains((Integer) withVotes);
            }
           return withVotes == this.votes;

becomes

            Set<Integer> menu = getVoteCountMenu();
            return menu == null ? withVotes == this.votes : menu.contains(withVotes);

Reply:
Good, I'll do that.